python/feathub/metric_stores/metric.py

```diff
@@ -192,3 +191,125 @@ def from_json(cls, json_dict: Dict) -> "Ratio":
             filter_expr=json_dict["filter_expr"],
             window_size=timedelta(seconds=json_dict["window_size_sec"]),
         )
+
+
+class CountMap(Metric):
+    """
+    A map-typed metric that maps each unique feature value to the number of occurrences
+    of this value.
+    """
+
+    def __init__(
+        self,
+        filter_expr: Optional[str] = None,
+        window_size: timedelta = timedelta(seconds=0),
+    ):
+        """
+        :param filter_expr: Optional. If it is not None, it represents a partial FeatHub
+                            expression which evaluates to a boolean value. The partial
+                            Feathub expression should be a binary operator whose left
+                            child is absent and would be filled in with the host feature
+                            name. For example, "IS NULL" will be enriched into
+                            "{feature_name} IS NULL". Only features that evaluate this
+                            expression into True will be considered when computing the
+                            metric.
+        :param window_size: The time range to compute the metric. It should be zero or
+                            a positive time span. If it is zero, the metric will be
+                            computed from all feature values that have been processed
+                            since the Feathub job is created.
+        """
+        super().__init__("count_map", window_size)
+        self.filter_expr = filter_expr
+
+    def get_tags(self) -> OrderedDict:
+        tags = super(CountMap, self).get_tags()
+        tags["count_map_key"] = "null"
```

Comment:
Do we need to have `count_map_key` as part of this metric's tags?

Note that this tag is not documented for the CountMetric in metrics.md. 

It is mentioned in the Java doc of `PrometheusSinkFunction` that `any label whose value is "null" in the column's comment should be updated to use the given key as the label value`. Thus, I am not sure we still need to explicitly specify `count_map_key=null` anywhere in FeatHub. It might be simpler to remove this string entirely from FeatHub.